goosebit/updates/swdesc.py

```diff
@@ -68,9 +71,14 @@ async def parse_file(file: Path):
 async def parse_remote(url: str):
     async with httpx.AsyncClient() as c:
         file = await c.get(url)
-        async with aiofiles.tempfile.NamedTemporaryFile("w+b") as f:
+        artifacts_dir = Path(config.artifacts_dir)
+        tmp_file_path = artifacts_dir.joinpath("tmp", ("".join(random.choices(string.ascii_lowercase, k=12)) + ".tmp"))
+        await tmp_file_path.parent.mkdir(parents=True, exist_ok=True)
+        async with await open_file(tmp_file_path, "w+b") as f:
             await f.write(file.content)
-            return await parse_file(Path(str(f.name)))
+        file_data = await parse_file(Path(str(f.name)))
+        await tmp_file_path.unlink()
```

Comment:
Move temp file deletion in a finally block to ensure that it always gets deleted.